packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -285,16 +303,23 @@ export function DocSearchModal({
                         .map(transformItems)
                         .map((groupedHits) =>
                           groupedHits.map((item) => {
+                            let parent: InternalDocSearchHit | null = null;
+
+                            const potentialParent = groupedHits.find(
+                              (siblingItem) =>
+                                siblingItem.type === 'lvl1' &&
+                                siblingItem.hierarchy.lvl1 ===
+                                  item.hierarchy.lvl1
+                            ) as InternalDocSearchHit;
```

Comment:
This lies a little bit less with the type:

```suggestion
                            ) as InternalDocSearchHit | undefined;
```